### PR TITLE
refactor portal login route

### PR DESCRIPTION
**Pull request by mariamisa**
relates #144

### CODE REVIEW

File: client/src/components/container/portal/LogInPage/index.js [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
+    resetFields();
   };
 
   render() {
-    const { isLoade, error, msg } = this.state;
+    const { isLoade, error, msg, isFinish } = this.state;
+    const { formRef, onReset, onFinish } = this;
+    if (isFinish) {
+      onReset();
+    }
```
Comment (AhmedSafi97) [inline client/src/components/container/portal/LogInPage/index.js:56]: This should be in the catch block, since we need to reset the form in case of an incorrect pin code.
Also, setting it here will throw this error after submitting incorrect pin code **Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.**

File: client/src/components/container/portal/LogInPage/index.js [hunk 2]
```diff
@@ -1,63 +1,71 @@
...
+    this.setState({ isLoade: true, isFinish: true });
+    try {
+      const { data } = await axios.post('/api/v1/portal/login', { pinCode });
+      push('/portal/attendance');
+      message.success(data.msg, 10);
+    } catch ({
+      response: {
+        data: { msg },
+      },
+    }) {
```
Comment (AhmedSafi97) [inline client/src/components/container/portal/LogInPage/index.js:32]: don't destructure the error object, check if it's a response error or a request error or something else, **See the pattern that we are using**  
https://github.com/GSG-G8/events-booker/blob/b09f6c9bc2119508a2336d14a0457865ee6012fd/client/src/components/container/user/Registeration/index.js#L38-L47

File: client/src/components/container/portal/LogInPage/index.js [hunk 3]
```diff
@@ -1,63 +1,71 @@
...
 import { Form, Input, Button, Spin, Alert, message } from 'antd';
 import axios from 'axios';
-
+import propTypes from 'prop-types';
 import './style.css';
 
-export class PortalLogin extends Component {
+export default class PortalLogin extends Component {
   state = {
     isLoade: false,
```
Comment (AhmedSafi97) [inline client/src/components/container/portal/LogInPage/index.js:9]: fix this typo please :sweat_smile:  , **"isLoaded"**

File: client/src/components/container/portal/LogInPage/index.js [hunk 4]
```diff
@@ -1,63 +1,71 @@
...
-    pinCode: '',
-    serverError: '',
+    isFinish: false,
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:10]: **there is no need for it**

File: client/src/components/container/portal/LogInPage/index.js [hunk 5]
```diff
@@ -1,63 +1,71 @@
...
-          });
-        }
-      );
+  onReset = () => {
+    const {
+      formRef: {
+        current: { resetFields },
+      },
+    } = this;
+    resetFields();
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:48]: there is no need to make a separate function here, add attribute ref to the Form as the following : 

**- define ref inside `render()` as the following :**

`const refInput =  React.createRef();
`

- **then :**

```
                         <Form
                          ref={refInput}
                          onFinish={(values) => {
                            const {
                              current: { resetFields },
                            } = refInput;
                            onFinish(values, resetFields);
                          }}
                          className="main-form" >
```

File: client/src/components/container/portal/LogInPage/index.js (hunk 2 above)
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:32]: reset input field should be inside catch : 
```suggestion
  onFinish = async ({ pinCode }, resetFields) => {
    const {
      props: {
        history: { push },
      },
    } = this;
    this.setState({ isLoade: true, isFinish: true });
    try {
      const { data } = await axios.post('/api/v1/portal/login', { pinCode });
      push('/portal/attendance');
      message.success(data.msg, 10);
    }  catch (err) { 
     let error; 
     resetFields();
     if (err.response) { 
       error = err.response.data.msg; 
     } else { 
       error = 'Something went wrong, please try again later'; 
     } 
     this.setState({ error, loading: false }); 
   } 
```

File: server/app.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ const app = express();
 app.disabled('x-powered-by');
 
 app.use(compression());
+app.use(express.urlencoded({ extended: false }));
```
Comment (AlaaTaima) [inline server/app.js:14]: **why did you add it ??**
